esiclient/v1/trunk.py

```diff
@@ -0,0 +1,147 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import logging
+
+from osc_lib.command import command
+from osc_lib.i18n import _
+
+from esiclient import utils
+
+ACTIVE = 'active'
+ADOPT = 'adopt'
+MANAGEABLE = 'manageable'
+
+
+class List(command.Lister):
+    """List existing trunk ports and subports"""
+
+    log = logging.getLogger(__name__ + ".List")
+
+    def get_parser(self, prog_name):
+        parser = super(List, self).get_parser(prog_name)
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)" % parsed_args)
+
+        neutron_client = self.app.client_manager.network
+        trunks = neutron_client.trunks()
+
+        data = []
+        for trunk in trunks:
+            trunk_port = neutron_client.get_port(trunk.port_id)
+            network_names, port_names, fixed_ips \
+                = utils.get_full_network_info_from_port(
+                    trunk_port, neutron_client)
+            data.append([trunk.name,
+                         "\n".join(port_names),
+                         "\n".join(network_names)])
+
+        return ["Trunk", "Port", "Network"], data
+
+
+class Create(command.ShowOne):
+    """Create trunk port with subports"""
+
+    log = logging.getLogger(__name__ + ".Create")
+
+    def get_parser(self, prog_name):
+        parser = super(Create, self).get_parser(prog_name)
+        parser.add_argument(
+            "name",
+            metavar="<name>",
+            help=_("Name of trunk"))
+        parser.add_argument(
+            "--network",
+            metavar="<network>",
+            help=_("Name or UUID of the primary network"))
+        parser.add_argument(
+            '--secondary-networks',
```

Comment:
is it safe to assume that secondary networks will be attached as tagged networks on the switchport, and the primary network will be the native network?

Reply:
Yep - I'll rename these parameters to be clearer.